packages/eslint-mdx/src/helpers.ts

```diff
@@ -112,23 +112,34 @@ export const restoreNodeLocation = <T>(
   const start = range[0] + offset
   const end = range[1] + offset
 
+  const restoredStartLine = startLine + startLoc.line
+  const restoredEndLine = startLine + endLoc.line
+
   return Object.assign(node, {
     start,
     end,
     range: [start, end],
     loc: {
       start: {
-        line: startLine + startLoc.line,
-        column: startLoc.column,
+        line: restoredStartLine,
+        column: startLoc.column + (restoredStartLine === 1 ? offset : 0),
       },
       end: {
-        line: startLine + endLoc.line,
-        column: endLoc.column,
+        line: restoredEndLine,
+        column: endLoc.column + (restoredEndLine === 1 ? offset : 0),
       },
     },
   })
 }
 
+export const arrayify = <T, R = T extends Array<infer S> ? S : T>(
+  ...args: T[]
+) =>
+  args.reduce<R[]>((arr, curr) => {
+    arr.push(...(Array.isArray(curr) ? curr : curr == null ? [] : [curr]))
+    return arr
```

Comment:
I'd avoid nesting terneries if possible, sometimes using multiple returns can be clearer

```suggestion
    if (Array.isArray(curr)) return arr.concat(curr)
    if (curr == null) return arr
    return arr.concat([curr])
```

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md

Reply:
I think it's OK for such simple statement.

Reply:
> https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-spread.md

:man_shrugging: okay, then

```suggestion
    if (Array.isArray(curr)) return [...arr, ...curr]
    if (curr == null) return arr
    return [...arr, curr]
```

Reply:
This solution iterates the original whole arr everytime, I would still prefer `push` version.

Reply:
> I think it's OK for such a simple statement.

I'm somewhat surprised it wasn't flagged by the linter, many configs have https://eslint.org/docs/rules/no-nested-ternary enabled.

Reply:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-nested-ternary.md

Reply:
Okay?
This code doesn't follow the style guide in either https://eslint.org/docs/rules/no-nested-ternary or https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-nested-ternary.md

Reply:
Seems been disabled by `eslint-plugin-prettier` again due to parens. 🤣

And it follows `the nested ternary is only one level`